Controller/CoreController.php

```diff
@@ -158,10 +155,25 @@ protected function getSearchHandler()
      */
     protected function getBaseTemplate()
     {
-        if ($this->getRequest()->isXmlHttpRequest()) {
+        if ($this->getCurrentRequest()->isXmlHttpRequest()) {
             return $this->getAdminPool()->getTemplate('ajax');
         }
 
         return $this->getAdminPool()->getTemplate('layout');
     }
+
+    /**
+     * Get the request object from the container.
+     *
+     * @return Request
+     */
+    private function getCurrentRequest()
+    {
+        // NEXT_MAJOR: simplify this when dropping sf < 2.4
```

Comment:
Someone might be using the methods from an action that does not have the request. They will have to inject the request too. Plus, it's a signature change, which means more deprecations.